### ISSUE TITLE
Finetune Albert on MovieReview dataset

### ISSUE THREAD

**Issue by PhilippMarquardt** [kpe/bert-for-tf2#53]
Hi!
I tried finetuning the albert base/large model on the MovieReview dataset that is used in the bert example.  
The model is created like this:
 def create_model(max_seq_len):

    albert_model_name = "albert_base"
    albert_dir = bert.fetch_tfhub_albert_model(albert_model_name, ".models")
    model_params = bert.albert_params(albert_dir)
    l_bert = bert.BertModelLayer.from_params(model_params, name="albert")    
    input_ids      = keras.layers.Input(shape=(max_seq_len,), dtype='int32', name="input_ids")
    #token_type_ids = keras.layers.Input(shape=(max_seq_len,), dtype='int32', name="token_type_ids")
    #output         = l_bert([input_ids, token_type_ids])
    output         = l_bert(input_ids)

    print("bert shape", output.shape)
    cls_out = keras.layers.Lambda(lambda seq: seq[:, 0, :])(output)
    cls_out = keras.layers.Dropout(0.5)(cls_out)
    logits = keras.layers.Dense(units=1024, activation="tanh")(cls_out)
    logits = keras.layers.Dropout(0.5)(logits)
    logits = keras.layers.Dense(units=2, activation="softmax")(logits)

    # model = keras.Model(inputs=[input_ids, token_type_ids], outputs=logits)
    # model.build(input_shape=[(None, max_seq_len), (None, max_seq_len)])
    model = keras.Model(inputs=input_ids, outputs=logits)
    model.build(input_shape=(None, max_seq_len))
    model.compile(optimizer=keras.optimizers.Adam(),
                loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
                metrics=[keras.metrics.SparseCategoricalAccuracy(name="acc")])
    
    
    # load the pre-trained model weights
    bert.load_albert_weights(l_bert, albert_dir)

    

    model.summary()

    return model

I've used `from bert.tokenization.albert_tokenization import FullTokenizer` for tokenization

Everything else is like in the provided bert example.

When executing the training loop, the accuracy stays at 50% and the loss doesn't really change from .7

Has anybody successfully finetuned any pretrained albert model on the MovieReview dataset? If yes, what am I doing wrong? Thanks in advance!

**Comment by kpe** [comment]
@PhilippMarquardt - it might be, that your learning rate is too high. Try different learning rates, i.e. somewhere around 1e-5, or check the examples under https://github.com/kpe/bert-for-tf2/tree/master/examples

**Comment by raviolli** [comment]
Any luck with this. I set the LR to 5e-5 but nothing... same problem

**Comment by eschibli** [comment]
I have the same issue with outputs never evolving away from 0.5 and loss remaining at 0.7 regardless of learning rate or optimizer. Could you confirm that it is working and post an example if possible?

**Comment by Ritika2001** [comment]
I am having the same issue. The model is predicting everything as 1 class.

**Comment by SebastienKeroack** [comment]
After saving the model, I can see this:
`
WARNING:tensorflow:Skipping full serialization of Keras layer <tensorflow.python.keras.layers.embeddings.Embedding object at 0x000001F307B6AC88>, because it is not built.
`

This is my build:
`
	if(batch_size == 1): INPUT = tf.keras.layers.Input(batch_shape = (1,) + shape_inputs)
	else               : INPUT = tf.keras.layers.Input(shape       =        shape_inputs)

	model_name   = 'albert_base'
	model_dir    = bert.fetch_tfhub_albert_model(model_name, os.path.join('test', 'nlp_5'))
	model_params = bert.albert_params(model_name)

	l_bert = bert.BertModelLayer.from_params(model_params, name='albert')
	X = l_bert(INPUT)

	# http://jalammar.github.io/a-visual-guide-to-using-bert-for-the-first-time/
	# X = ['All sentences', 'Only the first position: [CLS]', 'All hidden unit outputs']
	X = tf.keras.layers.Lambda(lambda X: X[:, 0, :])(X)
	X = tf.keras.layers.Dropout(0.5)(X)
	X = tf.keras.layers.Dense(units=768, activation='tanh')(X)
	X = tf.keras.layers.Dropout(0.5)(X)
		
	OUTPUT = tf.keras.layers.Dense(n_classes, activation='softmax')(X)

	model = tf.keras.models.Model(INPUT, OUTPUT, name='model_sentiment')
	model.build(input_shape=(None,) + shape_inputs)

	skipped_weight_value_tuples = bert.load_albert_weights(l_bert, model_dir)
	assert(0, len(skipped_weight_value_tuples))

	optimizer = tf.keras.optimizers.Adam(learning_rate=1e-3)

	model.compile(
		optimizer,
		loss=tf.keras.losses.CategoricalCrossentropy(from_logits=False),
		metrics=[tf.keras.metrics.CategoricalAccuracy()]
	)
`
